lib/facility_factory.rb

```diff
@@ -0,0 +1,49 @@
+class FacilityFactory
+
+    def create_facilities(facility_data)
+        facility_data.map do |facility|
+            if facility[:state] == 'CO'
+                facility_details = {
+                    name: facility[:dmv_office],
+                    address: format_address(facility[:address_li], facility[:address__1], 
+                    facility[:city], facility[:state], facility[:zip]),
```

Comment:
Just missing one thing -- services! We want to read in the services from the external data so that the facility can perform the appropriate tasks